Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPILocalizationCocoa.mm

```diff
@@ -51,8 +52,9 @@
 {
     // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/i18n/getMessage
 
-    // FIXME: <https://webkit.org/b/246488> Retrieve localized dictionary from the WebExtensionContextProxy.
-    _WKWebExtensionLocalization *localizedDictionary;
+    NSDictionary *localizationDictionary = extensionContext().localizationDictionary();
+    _WKWebExtensionLocalization *localization = [[_WKWebExtensionLocalization alloc] initWithRegionalLocalization:localizationDictionary languageLocalization:nil defaultLocalization:nil withBestLocale:nil uniqueIdentifier:extensionContext().uniqueIdentifier()];
```

Comment:
It would be better to store `_WKWebExtensionLocalization` on the context, and access it here instead of the dictionary so we are not making `_WKWebExtensionLocalization` every time.